roles/vdm/tasks/deploy.yaml

```diff
@@ -1,99 +1,100 @@
+        paths: "{{ ORCHESTRATION_TOOLING_INSTALL_MANIFESTS_DIRECTORY }}"
+        patterns: "*.yaml"
+      register: manifests
+    - name: deploy - Gather SAS Viya deployment resource files
+      find:
+        paths: "{{ ORCHESTRATION_TOOLING_INSTALL_MANIFESTS_DIRECTORY }}"
+        patterns: "*.yaml"
+        contains: "^kind: SASDeployment"
+      register: deployment_manifests
+      failed_when:
+        - ( deployment_manifests.files | length ) != 1
+    - name: deploy - Apply supporting resources
+      community.kubernetes.k8s:
+        src: "{{ item.path }}"
+        state: present
+        namespace: "{{ NAMESPACE }}"
+        kubeconfig: "{{ KUBECONFIG }}"
+      with_items:
+        - "{{ manifests.files | difference(deployment_manifests.files) }}"
+    - name: orchestration - Create orchestration tooling work directory
+      file:
+        path: "{{ ORCHESTRATION_TOOLING_DIRECTORY }}/work"
+        state: directory
+    - name: deploy - Deploy SAS Viya
+      environment:
+        PATH: "{{ ORCHESTRATION_TOOLING_PATH }}"
+        KUBECONFIG: "{{ KUBECONFIG }}"
+        WORK_DIRECTORY: "{{ ORCHESTRATION_TOOLING_DIRECTORY }}/work"
+      command: |
+        orchestration deploy
+             --namespace "{{ NAMESPACE }}"
+             --sas-deployment-cr "{{ item.path }}"
+      args:
+        chdir: "{{ ORCHESTRATION_TOOLING_DIRECTORY }}"
+      with_items:
+        - "{{ deployment_manifests.files }}"
+      when:
+        - deployment_tooling == "docker"
+    - name: deploy - Deploy SAS Viya
+      community.docker.docker_container:
+        name: orchestration
+        image: "{{ V4_CFG_CR_HOST }}/{{ ORCHESTRATION_IMAGE }}"
+        volumes:
+          - "{{ ORCHESTRATION_TOOLING_INSTALL_MANIFESTS_DIRECTORY }}:/manifests"
+          - "{{ KUBECONFIG }}:/config/kubeconfig"
+        detach: false
+        cleanup: yes
+        user: "{{ UID_GID }}"
+        env:
+          KUBECONFIG: "/config/kubeconfig"
+          WORK_DIRECTORY: "/tmp/work"
+        command:
+          - "deploy"
+          - "--namespace {{ NAMESPACE }}"
+          - "--sas-deployment-cr {{ item.path | replace(ORCHESTRATION_TOOLING_DIRECTORY, '/') }}"
+      with_items:
+        - "{{ deployment_manifests.files }}"
+      when:
+        - deployment_tooling == "ansible"
+  when:
+    - V4_DEPLOYMENT_OPERATOR_ENABLED == False
   tags:
     - install
     - update
+    - cas-onboard
 
-- name: Remove Viya
-  community.kubernetes.k8s:
-    state: absent
-    src:  "{{ DEPLOY_DIR }}/site.yaml"
-    wait: false
-    kubeconfig: "{{ KUBECONFIG }}"
-  register: result
-  ignore_errors: yes
+- name: deploy - Remove Viya
+  block:
+    - name: deploy - Remove Viya
+      environment:
+        KUBECONFIG: "{{ KUBECONFIG }}"
+      ansible.builtin.shell: |
+        kubectl -n {{ NAMESPACE }} delete pgclusters.webinfdsvr.sas.com --all || true
+        kubectl -n {{ NAMESPACE }} delete -f {{ ORCHESTRATION_TOOLING_UNINSTALL_MANIFEST }}
+      ignore_errors: yes
```

Comment:
Why are we switching between the community module and the basic shell command here? Can you not use the `with_items` element and have the src be driven from that?

Reply:
Missed this question, I actually reverted this back. Initially I was running into issues cleaning PG resources so I tried this out, now that it's resolved this has been changed back to the community module.